## Dimensionality Reduction of Data
- Now that the original movies_df dataset has been cleaned (removal/substitution of NaN values) and expanded (via pd.Dummies()), the resulting dataframe is large, with many variables, and needs to be reduced down to important data columns.

In [1]:
# Import data
%store -r movies_clean

In [2]:
# Describe data
movies_clean.describe()

title        tagline       revenue        budget  \
count  201133.000000  201133.000000  2.011330e+05  2.011330e+05   
mean      203.736721     203.204959  4.114268e+08  1.307756e+08   
std       127.433804     127.925826  3.276730e+08  4.289526e+07   
min         0.000000       0.000000  0.000000e+00  0.000000e+00   
25%        91.000000      81.000000  2.116432e+08  1.000000e+08   
50%       197.000000     195.000000  3.566134e+08  1.300000e+08   
75%       319.000000     297.000000  5.240287e+08  1.500000e+08   
max       436.000000     435.000000  2.787965e+09  3.800000e+08   

                  id  original_language       overview   release_date  \
count  201133.000000      201133.000000  201133.000000  201133.000000   
mean    36435.373623           0.028518     202.482213     204.921455   
std     70659.064671           0.169644     126.433443     115.751351   
min        12.000000           0.000000       0.000000       0.000000   
25%       956.000000           0.000000      81.000000     108.000000   
50%      2454.000000           0.000000     208.000000     238.000000   
75%     37710.000000           0.000000     302.000000     279.000000   
max    417859.000000           2.000000     436.000000     412.000000   

             runtime    status  ...  spoken_lang_বাংলা  spoken_lang_ਪੰਜਾਬੀ  \
count  201133.000000  201133.0  ...           201133.0            201133.0   
mean      127.453595       0.0  ...                0.0                 0.0   
std        21.486860       0.0  ...                0.0                 0.0   
min        76.000000       0.0  ...                0.0                 0.0   
25%       115.000000       0.0  ...                0.0                 0.0   
50%       126.000000       0.0  ...                0.0                 0.0   
75%       144.000000       0.0  ...                0.0                 0.0   
max       201.000000       0.0  ...                0.0                 0.0   

       spoken_lang_தமிழ்  spoken_lang_తెలుగు  spoken_lang_ภาษาไทย  \
count      201133.000000            201133.0        201133.000000   
mean            0.001034                 0.0             0.024809   
std             0.032142                 0.0             0.155544   
min             0.000000                 0.0             0.000000   
25%             0.000000                 0.0             0.000000   
50%             0.000000                 0.0             0.000000   
75%             0.000000                 0.0             0.000000   
max             1.000000                 0.0             1.000000   

       spoken_lang_ქართული  spoken_lang_广州话 / 廣州話  spoken_lang_日本語  \
count             201133.0          201133.000000    201133.000000   
mean                   0.0               0.012181         0.037413   
std                    0.0               0.109694         0.189772   
min                    0.0               0.000000         0.000000   
25%                    0.0               0.000000         0.000000   
50%                    0.0               0.000000         0.000000   
75%                    0.0               0.000000         0.000000   
max                    0.0               1.000000         1.000000   

       spoken_lang_普通话  spoken_lang_한국어/조선말  
count    201133.000000        201133.000000  
mean          0.046855             0.005946  
std           0.211328             0.076883  
min           0.000000             0.000000  
25%           0.000000             0.000000  
50%           0.000000             0.000000  
75%           0.000000             0.000000  
max           1.000000             1.000000  

[8 rows x 15007 columns]

In [3]:
# Assess shape of data
movies_clean.shape

(201133, 15007)

*Observations*: A dataset with 201133 instances (rows) and 15007 variables (columns) per instance is quite large, and slow to load, as well as has the risk of having too much dimension for a transferrable resuly.. This dataset thus needs to be reduced in variables.